### PR TITLE
Migrate CI to github actions

### LINKED ISSUE #6453: Get ready for `circleci/jruby` deprecation
I received this email from CircleCI a while ago:

JRubyDeprecation

Apparently `circleci/jruby` will stop receiving updates soon, so we'll need to build our own base image.

Since we're going to need to do this, maybe we can take the chance and migrate to Github Actions + [Github Container Registry](https://github.blog/2020-09-01-introducing-github-container-registry/).

### LINKED ISSUE #6493: Prepare (if necessary) for CircleCI + docker changes
I just got this email:

> Hi there,
>
> On November 1st, Docker Hub will begin limiting anonymous image pulls. We want to make sure you know how you might be impacted and what you can do to avoid interruptions to your workflow.
>
> Adding Docker authentication to your pipeline config is the easiest way to avoid any service disruptions. If you use the Docker executor or pull Docker images when using the machine executor on CircleCI, we encourage you to authenticate. Because the anonymous API rate limits are based on IP addresses, they will impact CircleCI cloud customers. Authenticated users get higher per-user rate limits, regardless of IP.
>
> We are currently working on a partnership with Docker to minimize the impact of this change for our users and will share more details as we get them.
>
> For more information or to leave a question for us, please head over to Discuss or contact support.
>
> Thanks and happy building,
>
> - The CircleCI team 

Just copying it here to make sure we don't forget about it.

### LINKED ISSUE #6559: Migrate to Github Actions + Github Container Registry
Issues with CircleCI start piling up.

They are [removing jruby images](https://github.com/activeadmin/activeadmin/issues/6453), [limiting docker pulls](https://github.com/activeadmin/activeadmin/issues/6493), and now they're limiting our concurrency credits making our CI slow.

Let's migrate to Github Actions.

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
The base github actions environment essentially contains everything we need (browers, node), and the extra stuff like rubies are better easy to setup, so we don't need to maintain any base docker images. Of course that's a bit less deterministic since we can't "freeze" the github actions environment like we do with docker images, but things are unlikely to break because of this and if it does... we'll deal with it.

Closes #6559.
Closes #6453.
Closes #6493.

### CODE REVIEW

**Comment by deivid-rodriguez** [comment]
This is passing now, rebased on top of #6569. Also, I partially reverted #6563 because the same failures in javascript scenarios appeared again, so they were not related to running in parallel.

**Comment by deivid-rodriguez** [comment]
Also, the build is fast now! 2-3 minutes on non-jruby rubies, and 9-10 on jrubies, with great concurrency between jobs as well! :tada:

**Comment by deivid-rodriguez** [comment]
Just realized that I forgot to migrate all the linting stuff. I'll do it shortly.

**Comment by deivid-rodriguez** [comment]
It should be ready now!

**Comment by deivid-rodriguez** [comment]
CircleCI has already been disconnected.

**Comment by deivid-rodriguez** [comment]
Codeclimate code coverage checks show as pending because CI didn't run on the latest commit in master (https://github.com/activeadmin/activeadmin/commit/7177af97cc157d7210a6bfce2cfc9362478041ae), so codeclimate never receive the information to calculate the base coverage for the PR.

There's nothing we can do about that, it should get auto healed with the next PR that we merge.

**Review by javierjulio — APPROVED** [review]
Looks great, thanks David for taking care of this! 🙌🏻

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  test:
+    runs-on: ubuntu-20.04
+
+    timeout-minutes: 15
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        ruby: [2.7.1, 2.6.6]
```
[the remark this answers is not in the document]
Comment (deivid-rodriguez) [reply .github/workflows/ci.yaml:88]: Yay!

**Comment by deivid-rodriguez** [comment]
Thanks @javierjulio!!

**Pull request merged**